File: gabriel/Problema0367/Problema0367.java
package Problema0367;

class Problema0367{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
  
      int matriz[][] = new int[4][4];
      boolean pontoDeSela = false;
  
      for(int i=0; i<4; i++){
        for(int j=0; j<4; j++){
          System.out.println("Digite o valor da linha "+i+ " e da coluna "+j);
          matriz[i][j] = Integer.valueOf(input.nextLine());
        }
      }
      
      
      for(int i=0; i<4; i++){
          int menorDaLinha = matriz[i][0]; int indiceColuna = 0; int indiceLinha = 0;
          for(int j=1; j<4; j++){
              if(menorDaLinha > matriz[i][j]){
                  menorDaLinha = matriz[i][j];
                  indiceColuna = j;
                  indiceLinha = i;
              }
          }
          int k;
          for(k=0; k<4; k++){
              if(menorDaLinha < matriz[k][indiceColuna]){
                  break;
              }
          }
          
          if(k == 4){
              pontoDeSela = true;
              System.out.println("Existe ponto de sela na posição: linha: " +indiceLinha+ " coluna: "+indiceColuna );
              System.out.println("O valor é "+ menorDaLinha);
              
          }
    
      }
      
      if(pontoDeSela == false){
          System.out.println("Não existe ponto de sela na matriz");
      }
     }
}

File: gabriel/Problema0367/Problema0367.py
matriz  = [[0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]
pontoSela = False
 
for i in range(4):
    for j in range(4):
        matriz [i][j] = int(input("Digite o valor da linha "+str(i)+" e da coluna "+str(j) + ": "));  


for i in range(4):
       
    menorDaLinha = matriz[i][0]
    indiceColuna = 0; indiceLinha = 0
    for j in range(1, 4):
        if (menorDaLinha > matriz[i][j]):
            menorDaLinha = matriz[i][j]
            indiceColuna = j
            indiceLinha = i
 
        
    k = 0
    for k in range(4):
 
            
        if (menorDaLinha < matriz[k][indiceColuna]):
            break
        k += 1
 
    if (k == 4):
        print("Existe um ponto de sela na posição da linha " +str(indiceLinha)+" e da coluna "+str(indiceColuna))
        print("O valor é "+ str(menorDaLinha))
        pontoSela = True

if(pontoSela == False):
    print("Não existe ponto de sela na matriz")

            
 
    
    
 


   
    
    
 
